File: python/src/azex/types/usage.py
"""Types for the usage resource."""

from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel


class ModelUsageStat(BaseModel):
    model: str
    request_count: int
    total_tokens: int
    prompt_tokens: int
    completion_tokens: int
    total_cost: str


class UsageStats(BaseModel):
    request_count: int
    total_tokens: int
    prompt_tokens: int
    completion_tokens: int
    total_cost: str
    by_model: List[ModelUsageStat]
    from_date: Optional[str] = None
    to_date: Optional[str] = None


class UsageLog(BaseModel):
    uid: str
    model: str
    provider: Optional[str] = None
    prompt_tokens: int
    completion_tokens: int
    total_tokens: int
    cost: str
    latency_ms: Optional[int] = None
    status: str
    error: Optional[str] = None
    created_at: str
    is_stream: bool
    is_estimated: bool

File: typescript/src/types/usage.ts
export interface ModelUsageStat {
  model: string;
  request_count: number;
  total_tokens: number;
  prompt_tokens: number;
  completion_tokens: number;
  total_cost: string;
}

export interface UsageStats {
  request_count: number;
  total_tokens: number;
  prompt_tokens: number;
  completion_tokens: number;
  total_cost: string;
  by_model: ModelUsageStat[];
  from_date?: string;
  to_date?: string;
}

export interface UsageLog {
  uid: string;
  model: string;
  provider?: string;
  prompt_tokens: number;
  completion_tokens: number;
  total_tokens: number;
  cost: string;
  latency_ms?: number;
  status: string;
  error?: string;
  created_at: string;
  is_stream: boolean;
  is_estimated: boolean;
}

export interface UsageLogPage {
  items: UsageLog[];
  total: number;
  page: number;
  size: number;
  pages: number;
}
